cpp/src/table/table.cpp

```diff
@@ -29,7 +29,7 @@ table::table(table const& other) : _num_rows{other.num_rows()}
   CUDF_FUNC_RANGE();
   _columns.reserve(other._columns.size());
   for (auto const& c : other._columns) {
-    _columns.emplace_back(std::make_unique<column>(*c));
+    _columns.emplace_back(std::make_unique<column>(*c, cudf::get_default_stream()));
```

Comment:
Is this constructor ever used internally?

Reply:
Good catch. `table(table)` constructor is not internally used. 
It's only used in `cpp/benchmarks/stream_compaction/unique.cpp` (which could be replaced too).

Should we consider removing it?

Reply:
I think I want to see similarities between the design of columns and tables, so that probably leans towards keeping this. However, I could be swayed if there is an explicit API (free function?) that allows for table or column copying. If we decide to remove, I think we'd want an issue documenting this and to deprecate it before removing.

Reply:
My concern with having copy constructors for tables/columns is that we risk accidental copies as they can happen implicitly. On the other hand, we usually access these through unique_ptr, which would prevent copies.
I wonder if marking the copy ctors as explicit suffices here.

Reply:
added explicity and fixed the copies in benchmark

Reply:
Should we add a `stream` parameter for this constructor? Always using the default stream may be an issue.